### ISSUE TITLE
Bybit Stop Limit Order with TP and SL

### ISSUE THREAD

**Issue by calcifieddeveloper** [ccxt/ccxt#17541]
### Operating System

Ubuntu

### Programming Languages

python

### CCXT Version

3.0.61

### Description

Hi,
I am using Python 3.10.
I am trying to place a stop limit order with both TP and SL on bybits test net.
I have upgraded to a unified account.
The order goes though ok but without TP and SL.

Any ideas ?
Thanks for your time.

fetch Request: bybit POST https://api-testnet.bybit.com/v5/order/create RequestHeaders: {'Content-Type': 'application/json', 'X-BAPI-API-KEY': '35eW3QqJIXeVZLCgwW', 'X-BAPI-TIMESTAMP': '1681305003536', 'X-BAPI-RECV-WINDOW': '5000', 'X-BAPI-SIGN': '19613cf1617c3972a4f0b3ebf9022ca86f73edda264c1a53295ff6e78cb08b84', 'Referer': 'CCXT', 'User-Agent': 'python-requests/2.28.2', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} RequestBody: {"symbol":"MASKUSDT","side":"Buy","orderType":"Limit","category":"linear","qty":"92.4","price":"6.055","triggerDirection":1,"triggerPrice":"6.051","orderLinkId":"baf33591-7e90-45a9-be63-21be8b0ec550","basePrice":6.0,"tpTriggerBy":"LastPrice","slTriggerBy":"LastPrice","triggerBy":"LastPrice"}

fetch Response: bybit POST https://api-testnet.bybit.com/v5/order/create 200 ResponseHeaders: {'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '177', 'Connection': 'keep-alive', 'Date': 'Wed, 12 Apr 2023 13:10:03 GMT', 'X-Bapi-Limit': '10', 'X-Bapi-Limit-Status': '9', 'X-Bapi-Limit-Reset-Timestamp': '1681305003623', 'Ret_code': '0', 'Traceid': '4cec0bf33e403e6275a86027e4185a25', 'Timenow': '1681305003625', 'Server': 'Openresty', 'X-Cache': 'Miss from cloudfront', 'Via': '1.1 72d783f19e38dd5db23a648a65feb4f2.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'SYD62-P2', 'X-Amz-Cf-Id': 'fNTCtRQVaU9LrLBBif7jpqfCw32XWYDRM79KhXu8IveU_MgzV6C3NA=='} ResponseBody: {"retCode":0,"retMsg":"OK","result":{"orderId":"21770cd7-d292-42fb-ad84-8757c903971f","orderLinkId":"baf33591-7e90-45a9-be63-21be8b0ec550"},"retExtInfo":{},"time":1681305003625}
{'info': {'orderId': '21770cd7-d292-42fb-ad84-8757c903971f', 'orderLinkId': 'baf33591-7e90-45a9-be63-21be8b0ec550'}, 'id': '21770cd7-d292-42fb-ad84-8757c903971f', 'clientOrderId': 'baf33591-7e90-45a9-be63-21be8b0ec550', 'timestamp': None, 'datetime': None, 'lastTradeTimestamp': None, 'symbol': None, 'type': None, 'timeInForce': None, 'postOnly': None, 'reduceOnly': None, 'side': None, 'price': None, 'stopPrice': None, 'triggerPrice': None, 'amount': None, 'cost': None, 'average': None, 'filled': None, 'remaining': None, 'status': None, 'fee': None, 'trades': [], 'fees': []}

### Code

```
symbol = 'MASK/USDT:USDT'
amount = 92.4556213017751
price = 6.00
trigger_price = 6.051
entry_price = 6.055
takeprofit = 6.067
stoploss = 2

bybit.set_sandbox_mode(True)
bybit.options['defaultType'] = 'future'
bybit.options['enableUnifiedAccount'] = True
bybit.verbose = True
bybit.load_markets()

orderid = str(uuid.uuid4())

order_params = {
    'clientOrderId': orderid,
    'stopPrice': trigger_price,
    'basePrice': price,
    'stopLoss': stoploss,
    'takeProfit' : takeprofit,
    'tpTriggerBy': 'LastPrice',
    'slTriggerBy': 'LastPrice',
    'triggerBy': 'LastPrice',
    'triggerDirection': 1
}

order = bybit.create_unified_account_order(symbol, 'limit', 'buy', amount, entry_price, order_params)

```

**Comment by carlosmiei** [comment]
Hello @calcifieddeveloper,
You're mixing a lot of different "order types" here, if you want limit + tp and sl you should not provide `stopPrice`, `basePrice`, `triggerDirection`, only `takeProfit` and `stopLoss` inside params

**Comment by calcifieddeveloper** [comment]
Thanks @carlosmiei - I appreciate the response.

If I change the code to this I get the position opened with a SL and TP,  Not a stop limit order (or perhaps my questions are too noob :) ). 

symbol = 'MASK/USDT:USDT'
amount = 92.4556213017751
price = 6.00
trigger_price = 6.051
entry_price = 6.055
takeprofit = 6.067
stoploss = 2

bybit.set_sandbox_mode(True)
bybit.options['defaultType'] = 'future'
bybit.options['enableUnifiedAccount'] = True
bybit.verbose = True
bybit.load_markets()

orderid = str(uuid.uuid4())

order_params = {
    'clientOrderId': orderid,
    #'stopPrice': trigger_price,
    #'basePrice': price,
    'stopLoss': stoploss,
    'takeProfit' : takeprofit,
    #'tpTriggerBy': 'LastPrice',
    #'slTriggerBy': 'LastPrice',
    #'triggerBy': 'LastPrice',
    #'triggerDirection': 1
}

order = bybit.create_unified_account_order(symbol, 'limit', 'buy', amount, entry_price, order_params)


RequestBody: {"symbol":"MASKUSDT","side":"Buy","orderType":"Limit","category":"linear","qty":"92.4","price":"6.055","stopLoss":"2","takeProfit":"6.067","orderLinkId":"0d37d343-c4c5-4eed-9c92-441f1920cc3d"}

ResponseBody: 
{"retCode":0,"retMsg":"OK","result":{"orderId":"acf2ba98-844b-45a3-a937-e1abb58b2fc7","orderLinkId":"0d37d343-c4c5-4eed-9c92-441f1920cc3d"},"retExtInfo":{},"time":1681385669017}
 

So, My question using my example is.  How do I place a order for MASK/USDT:USDT on by bit so that 
1. When it gets to trigger price a limit order is placed for entryprice
2. Make sure when the order goes in it has a TP and SL for the values I specify.

Any ideas ?
Or am I being dimm.

**Comment by ttodua** [comment]
I think this is an old issue and no longer exists with the latest version (v5 bybit).
here is an example: https://docs.ccxt.com/#/README?id=stoploss-and-takeprofit-orders-attached-to-a-position
if anyone experiences any issues, let us know
